includes/class-wc-checkout.php

```diff
@@ -691,19 +691,34 @@ public function get_posted_data() {
 				// phpcs:disable WordPress.Security.NonceVerification.Missing
 				switch ( $type ) {
 					case 'checkbox':
-						$value = isset( $_POST[ $key ] ) ? 1 : '';
+						$value = isset( $field['default'] ) && ! empty( $field['default'] ) ? 1 : '';
+						if ( isset( $_POST[ $key ] ) ) {
+							$value = 1;
+						}
```

Comment:
I see quite a lot of repetition here, the code is basically the same in all cases except for the actual value processing. I think we could do somethink like this instead:

```php
if ( isset( $_POST[ $key ] ) ) {
  $value = $_POST[ $key ];
}
elseif isset( $field['default'] ) {
  $value = $field['default'];
}
else {
  $value = '';
}

if($value) {
  switch($type) {
    case 'checkbox':
      $value=1;
      break;
    case 'multiselect':
      $value = implode( ', ', wc_clean( wp_unslash( $value ) ) );
      break;
    case ...
  }
}
```